pycbc/inference/likelihood.py

```diff
@@ -545,6 +545,34 @@ def loglikelihood(self, **params):
         """
         return self._dist.logpdf([params[p] for p in self.variable_args])
 
+class TestEggbox(BaseLikelihoodEvaluator):
+    r"""The test distribution is an 'eggbox' function:
+    $$f(\Theta)=-\left[2+\prod_{i=1}^{n}\cos\left(\frac{\theta_{i}}{2}\right)\right]^{5}$$
+
+    The number of dimensions is set by the number of ``variable_args`` that are
+    passed.
+
+    Parameters
+    ----------
+    variable_args : (tuple of) string(s)
+        A tuple of parameter names that will be varied.
+    \**kwargs :
+        All other keyword arguments are passed to ``BaseLikelihoodEvaluator``.
+
+    """
+    name = "test_eggbox"
+
+    def __init__(self, variable_args, **kwargs):
+        # set up base likelihood parameters
+        super(TestEggbox, self).__init__(variable_args, **kwargs)
+        # set the lognl to 0 since there is no data
+        self.set_lognl(0.)
+
+    def loglikelihood(self, **params):
+        """Returns the log pdf of the eggbox function.
+        """
+        l = (2 + numpy.prod([numpy.cos(p/2.) for p in params.values()])) ** 5
+        return l
```

Comment:
Do `return (2 + numpy.prod(numpy.cos([params[p] for p in self.variable_args])))**5`. It can happen that `params` contains things besides the variable args (e.g., if a sampling transform is used, params will contain both the variable args and the sampling args). Also, it's slightly faster to call `numpy.cos` on the list of values rather than a list of `numpy.cos`.